flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -290,6 +295,8 @@ public TaskExecutor(
 		final ResourceID resourceId = taskExecutorServices.getUnresolvedTaskManagerLocation().getResourceID();
 		this.jobManagerHeartbeatManager = createJobManagerHeartbeatManager(heartbeatServices, resourceId);
 		this.resourceManagerHeartbeatManager = createResourceManagerHeartbeatManager(heartbeatServices, resourceId);
+
+		this.ioExecutor = Executors.newSingleThreadExecutor(new ExecutorThreadFactory("taskexecutor-io"));
```

Comment:
I think it would be better to move this executor to `TaskManagerServices` so that we have the option to configure it later if needed (e.g., number of threads). What do you think, @tillrohrmann?

Reply:
Hi @GJL, I agree with your proposal. Moving this `ioExecutor` to `TaskManagerServices` will make code configurable and clear.

Reply:
Thanks, you still need to adjust `TaskManagerServicesBuilder`. By default it can use the `TestingUtils.defaultExecutor()`